fix(ci): chain Release Images → Release Operator workflow trigger

## Problem

The Release Operator workflow has no explicit dependency on Release Images. Both independently trigger on `release: [released]`, and the only coupling is an **unbounded polling loop** in the "Wait on operand images" step that polls `docker manifest inspect` every 10 seconds with no timeout (relies on the job-level `timeout-minutes: 60`).

During the 3.3.1 release, Release Images failed (console plugin checkstyle — fixed in #8947), and the Release Operator wasted 60 minutes polling for images that would never appear before timing out. This happened twice ([30303505580](https://github.com/Apicurio/apicurio-registry/actions/runs/30303505580), [30342910505](https://github.com/Apicurio/apicurio-registry/actions/runs/30342910505)).

Additionally, the Release Images workflow has `timeout-minutes: 120` while the operator has `timeout-minutes: 60`, so even in a successful but slow build, the operator can time out before images are ready.

## Proposed fix

Replace the implicit polling with an explicit workflow dispatch trigger:

1. **`release-images.yaml`**: Add a final step that triggers the operator release on success:
   ```yaml
   - name: Trigger Release Operator
     if: success()
     run: gh workflow run release-operator.yaml -f tag=$RELEASE_VERSION
   ```

2. **`release-operator.yaml`**:
   - Remove the `release: [released]` trigger (to avoid double-firing)
   - Remove the "Wait on operand images" step (images are guaranteed to exist)
   - Keep `workflow_dispatch` as the only trigger

This eliminates the busy-wait resource waste, ensures correct ordering, and gives clean failure semantics — if images fail, the operator release simply doesn't start.

## Other 3.3.1 failures (re-run only, no code changes needed)

- **Release Artifacts** ([30303505967](https://github.com/Apicurio/apicurio-registry/actions/runs/30303505967)): Javadoc failed on Lombok-generated `VersionOutputBuilder` in CLI module, then Maven reactor hung until 90-min timeout. Re-run in progress.
- **Release SDKs** ([30303505836](https://github.com/Apicurio/apicurio-registry/actions/runs/30303505836)): `poetry publish` to PyPI hung for 30 minutes. Flaky — just needs a re-run.
- **Release Images**: Needs re-run with tag `3.3.1` (the fix from #8947 is merged). After this PR is merged, the re-run will also automatically trigger Release Operator.

## Further analysis

### Operator release dependency on Release Images

The Release Operator workflow has no explicit dependency on Release Images. Both independently trigger on `release: [released]`, and the only coupling is an **unbounded polling loop** in the "Wait on operand images" step:

```bash
until docker manifest inspect "$IMAGE" >& /dev/null; do
  echo "Waiting on $IMAGE"; sleep 10
done
```

This loop has no timeout — it relies on the job-level `timeout-minutes: 60` to kill it. Since the Release Images workflow has `timeout-minutes: 120`, in the worst case the operator times out before images are even ready, even in a successful run.

**Proposed improvement:** Replace the implicit polling with an explicit trigger. Release Images should dispatch the Release Operator workflow on success:

```yaml
# At the end of release-images.yaml, after all images are pushed
- name: Trigger Release Operator
  if: success()
  env:
    GH_TOKEN: ${{ secrets.ACCESS_TOKEN }}
  run: gh workflow run release-operator.yaml -f tag=$RELEASE_VERSION
```

Then in Release Operator:
- Remove the `release: [released]` trigger (to avoid double-firing)
- Remove the "Wait on operand images" step entirely (images are guaranteed to exist)
- Keep `workflow_dispatch` as the only trigger (for manual re-runs)

This eliminates the busy-wait resource waste, ensures correct ordering, and gives clean failure semantics — if images fail, the operator release simply doesn't start rather than timing out after 60 minutes.

*Note: `ACCESS_TOKEN` may need the `workflow` scope added to dispatch workflows via API.*

### Other 3.3.1 release workflow failures

The `release: [released]` event triggered 7 independent workflows. Beyond Release Images and Release Operator, two more failed:

#### Release Artifacts — cancelled (timeout)
**Run:** [30303505967](https://github.com/Apicurio/apicurio-registry/actions/runs/30303505967)
**Failed step:** `Maven Deploy` (hit `timeout-minutes: 90`)

Root cause: Javadoc generation failed on `cli/src/main/java/io/apicurio/registry/cli/VersionCommand.java` — it references `VersionOutput.VersionOutputBuilder`, a Lombok-generated inner class invisible to Javadoc:
```
error: cannot find symbol
  symbol:   class VersionOutputBuilder
  location: class VersionOutput
```
After the Javadoc error, Maven's reactor thread pool appears to have hung (~80 minutes of silence before the timeout killed it). This is already being re-run (run [30369129326](https://github.com/Apicurio/apicurio-registry/actions/runs/30369129326) in progress).

#### Release SDKs — cancelled (timeout)
**Run:** [30303505836](https://github.com/Apicurio/apicurio-registry/actions/runs/30303505836)
**Failed step:** `Publish to PyPI` in the Python SDK job (hit `timeout-minutes: 30`)

The `poetry publish --build` command hung for the full 30-minute timeout. Logs are too large to retrieve via API so the exact cause is unclear — likely a PyPI upload hang or slow dependency resolution. This workflow has a pattern of failures across releases (3.2.6 failed on TypeScript/npm, 3.3.0 and 3.2.5 were also cancelled). Has not been re-run yet.

#### Summary of 3.3.1 release status

| Workflow | Status | Issue |
|----------|--------|-------|
| Release Workflow | ✅ success | — |
| Release Release Notes | ✅ success | — |
| Update Website | ✅ success | — |
| Release Milestones | ✅ success | — |
| Release Images | ❌ failure | Console plugin checkstyle (fixed in #8947, needs re-run) |
| Release Operator | ❌ timeout | Cascading from Release Images (needs re-run after images) |
| Release Artifacts | ❌ timeout | Lombok Javadoc error in CLI module (re-run in progress) |
| Release SDKs | ❌ timeout | Python PyPI publish hang (needs re-run) |

--- a/.github/workflows/release-images.yaml
+++ b/.github/workflows/release-images.yaml
@@ -206,6 +206,12 @@ jobs:
               $TAGS \
               --platform linux/amd64,linux/arm64,linux/s390x,linux/ppc64le .
 
+      - name: Trigger Release Operator
+        env:
+          GH_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+        run: |
+          gh workflow run release-operator.yaml --repo "$GITHUB_REPOSITORY" -f tag="$RELEASE_VERSION"
+
   # Slack notification using reusable workflow
   notify-slack:
     name: Slack Notification

--- a/.github/workflows/release-operator.yaml
+++ b/.github/workflows/release-operator.yaml
@@ -1,4 +1,7 @@
 name: Release Operator
+# Triggered by Release Images workflow after all operand images are pushed,
+# ensuring correct ordering without a polling loop. Can also be dispatched
+# manually for re-runs.
 on:
   workflow_dispatch:
     inputs:
@@ -10,9 +13,6 @@ on:
         type: boolean
         default: true
 
-  release:
-    types: [ released ]
-
 permissions:
   contents: read
 
@@ -20,7 +20,7 @@ jobs:
 
   release-build:
     name: Build & Publish
-    if: github.repository_owner == 'Apicurio' && (github.event_name == 'workflow_dispatch' || startsWith(github.event.release.tag_name, '3.'))
+    if: github.repository_owner == 'Apicurio'
     runs-on: ubuntu-24.04
     timeout-minutes: 60
     outputs:
@@ -31,19 +31,11 @@ jobs:
     steps:
 
       - name: Fetch Release Details
-        if: github.event_name == 'workflow_dispatch'
         run: |
           touch release.json && curl "https://api.github.com/repos/$GITHUB_REPOSITORY/releases/tags/${{ github.event.inputs.tag }}" > release.json
           echo "RELEASE_VERSION=$(cat release.json | jq -r '.name')" >> $GITHUB_ENV
           echo "BRANCH=$(cat release.json | jq -r '.target_commitish')" >> $GITHUB_ENV
           echo "RUN_TESTS=${{ github.event.inputs.run_tests }}" >> $GITHUB_ENV
-
-      - name: Configure release event env
-        if: github.event_name != 'workflow_dispatch'
-        run: |
-          echo "RELEASE_VERSION=${{ github.event.release.name }}" >> $GITHUB_ENV
-          echo "BRANCH=${{ github.event.release.target_commitish }}" >> $GITHUB_ENV
-          echo "RUN_TESTS=true" >> $GITHUB_ENV
 
       - name: Download Source Code
         run: |
@@ -118,16 +110,6 @@ jobs:
         working-directory: registry/operator
         run: |
           make OPERAND_IMAGE_TAG=$RELEASE_VERSION ADDITIONAL_IMAGE_TAG=latest image-buildx-push
-
-      - name: Wait on operand images
-        working-directory: registry/operator
-        run: |
-          export REGISTRY_APP_IMAGE="$(make OPERAND_IMAGE_TAG=$RELEASE_VERSION VAR=REGISTRY_APP_IMAGE variable-get)"
-          until docker manifest inspect "$REGISTRY_APP_IMAGE" >& /dev/null; do echo "Waiting on $REGISTRY_APP_IMAGE"; sleep 10; done
-          export REGISTRY_UI_IMAGE="$(make OPERAND_IMAGE_TAG=$RELEASE_VERSION VAR=REGISTRY_UI_IMAGE variable-get)"
-          until docker manifest inspect "$REGISTRY_UI_IMAGE" >& /dev/null; do echo "Waiting on $REGISTRY_UI_IMAGE"; sleep 10; done
-          export REGISTRY_GITOPS_SYNC_IMAGE="$(make VAR=REGISTRY_GITOPS_SYNC_IMAGE variable-get)"
-          until docker manifest inspect "$REGISTRY_GITOPS_SYNC_IMAGE" >& /dev/null; do echo "Waiting on $REGISTRY_GITOPS_SYNC_IMAGE"; sleep 10; done
 
       - name: Build operator bundle
         working-directory: registry/operator

